qubes_config/global_config/basics_handler.py

```diff
@@ -208,6 +208,105 @@ def get_model(self) -> TraitSelector:
         return self.model
 
 
+class PreloadDispvmHandler(AbstractTraitHolder):
+    """Handler for preloaded disposables. Requires SpinButton widgets:
+    'basics_preload_dispvm'"""
+
+    def __init__(
+        self,
+        qapp: qubesadmin.Qubes,
+        widget: Gtk.SpinButton,
+        dependent_widget: Gtk.ComboBox,
+        is_dvm_template: Optional[Callable] = None,
+    ):
+        self.qapp = qapp
+        self.widget = widget
+        self.dependent_widget = dependent_widget
+        self.is_dvm_template = is_dvm_template
+        self.preload_dispvm_adjustment = Gtk.Adjustment()
+        self.preload_dispvm_adjustment.configure(0, 0, 50, 1, 10, 0)
+        self.widget.configure(self.preload_dispvm_adjustment, 0.1, 0)
+        self.widget.set_value(self.get_current_value())
+        self.dependent_widget.connect("changed", self.on_dependent_changed)
+        self.widget.set_sensitive(bool(self.get_dependent_value()))
+
+    def get_dependent_value(self):
+        treeiter = self.dependent_widget.get_active_iter()
+        if treeiter is not None:
+            model = self.dependent_widget.get_model()
+            value = model[treeiter][3]
```

Comment:
There's a bunch of helper functions for those models, using VMListModeller, I think it would be better to use them (maybe pass the modeler instead of the combobox? not sure) instead of assuming the model stays unchanged (this way if someone changes the VMListModeler, this will surprisingly explode)